### DISCUSSION TITLE
How can I add RWObj packages into Macad.OCCT

### DISCUSSION THREAD

**Discussion by qq614438892** [Macad3D/Macad3D#56] [Q&A]
How can I add RWObj packages into Macad.OCCT

**Comment by dipts** [comment]
RWObj is not a toolkit used by Macad, which implements its own OBJ exporter/importer. If features are missing they should be implemented there. If I know what is missing maybe I can help to find the right approach.

**Reply by qq614438892** [reply]
Thank you for your answer. I understand that Macad does not use the OCCT DataExchange module. So what if I want to read and write gltf and dae files? And then there's how to apply materials to the model

**Reply by dipts** [reply]
Macad uses OCCT for the import and export of IGES and STEP files. Unlike the toolkit for the mesh formats, these toolkits are not based on the Application Framework. The Application Framework is not used by Macad, and given that OBJ is a very simple format, using the TKRWMesh toolkit for import/export would introduce an extremely large number of weighty dependencies.

For supporting materials this would not help either. For the support of materials there is the need to include layers from the data model of Macad to export materials or to create layers by materials when importing around. The handling of materials in OBJ/MTL is very straightforward and should not be a problem.

**Reply by qq614438892** [reply]
![image](https://github.com/Macad3D/Macad3D/assets/88705209/65afe1fc-e32b-4ba1-859c-c00f6dc78fd3)
Why Parsing package StdPrs failed?How to make it successful.

**Reply by dipts** [reply]
CastXML compiles the headers of StdPrs using clang, this seems to have failed. You can find a log of the compilation in `.intermediate\Macad.Occt\castxml\StdPrs.log`. This file contains hints about what the compiler does not like.
